### PR TITLE
symbolic expressions

### PR DESCRIPTION
**Pull request by fajb**
Hello,

A domain of *weak* symbolic expressions for bincat.
Might be faster if there is some dead code - think pf.

### CODE REVIEW

**Comment by trou** [comment]
Hello Frédéric,
I think you forgot to add `symbexp.ml` to your repo before making the pull request.

File: ocaml/src/Makefile [hunk 1]
```diff
@@ -46,7 +47,7 @@ DIROPT = -I `ocamlfind query newspeak` -I fixpoint -I `ocamlfind query ocamlgrap
 
 
 CAMLC   =ocamlc $(DIROPT)
-CAMLOPT =ocamlopt -g -w Ael -warn-error +a-7-3-52-50-42 $(DIROPT)
+CAMLOPT =ocamlopt -g -w Ael  -bin-annot $(DIROPT)
```
Comment (trou) [inline ocaml/src/Makefile]: Any reason why you changed the compiler's flags ?

File: ocaml/src/main.ml [hunk 2]
```diff
@@ -30,8 +30,9 @@ let process (configfile:string) (resultfile:string) (logfile:string): unit =
   Register.clear();
   (* generating modules needed for the analysis *)
   let module Vector 	 = Vector.Make(Reduced_bit_tainting) in
-  let module Pointer 	 = Pointer.Make(Vector)		       in
-  let module Domain 	 = Reduced_unrel_typenv.Make(Pointer)  in
+  let module Ptr     = Symbexp.Make(Vector)            in
```
Comment (trou) [inline ocaml/src/main.ml]: Shouldn't that be
```ocaml
let module Vector = ...
```

File: ocaml/src/Makefile (hunk 1 above)
Comment (fajb) [inline ocaml/src/Makefile]: Yes, but stricter compiler flags  are probably desirable.
I installed zarith with opam and I get 
Warning 58: no cmx file was found in path for module Z, and its interface was not compiled with -opaque

Removing Warning 58 as error would be sufficient.

PS: I indeed forgot to include the relevant code...

File: ocaml/src/main.ml (hunk 2 above)
Comment (fajb) [inline ocaml/src/main.ml]: You're right.

**Comment by trou** [comment]
Merged with some patches in branch `symbexp`.

Also, if you continue to contribute, while we do not have contributing guidelines. Please use at least 3 chars for variable names :)

**Pull request closed without merging**